### PR TITLE
ui: add subscribe jobs link and modal

### PR DESCRIPTION
**Pull request by harunurhan**
* INSPIR-2395

### CODE REVIEW

File: ui/src/jobs/components/SubscribeJobsModalButton.jsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+          onCancel={this.onModalCancel}
+          onOk={this.onModalSubscribeClick}
+        >
+          <div>
+            {hasError && (
+              <div className="mb3">
+                <Alert
+                  type="error"
+                  showIcon
+                  description="Coud not subscribe, please try again."
```
Comment (drjova) [inline ui/src/jobs/components/SubscribeJobsModalButton.jsx]:
```suggestion
                  description="Could not subscribe, please try again."
```

File: ui/src/jobs/components/__tests__/SubscribeJobsModalButton.test.jsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    onModalOKClick();
+
+    expect(subscribeJobMailingList).toHaveBeenCalledWith({
+      email: 'harun@cern.ch',
+      firstName: 'Harun',
+      lastName: 'Urhan',
+    });
+    expect(wrapper.find(Modal).prop('visible')).toBe(false);
+  });
+});
```
Comment (drjova) [inline ui/src/jobs/components/__tests__/SubscribeJobsModalButton.test.jsx]: case for disabled Ok button if ``!(email && firstName && lastName)``?

File: ui/src/jobs/components/SubscribeJobsModalButton.jsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            <div>
+              To be notified via email of new jobs in High Energy Physics,
+              please fill in the following information:
+            </div>
+            <div className="mt3">
+              <Input
+                name="email"
+                placeholder="Email"
+                value={email}
+                onChange={this.onInputChange}
```
Comment (drjova) [inline ui/src/jobs/components/SubscribeJobsModalButton.jsx]: Can we validate emails? or we'll rely on the 3rd party to support this kind of feature

File: ui/src/jobs/components/__tests__/SubscribeJobsModalButton.test.jsx (hunk 2 above)
Comment (harunurhan) [reply ui/src/jobs/components/__tests__/SubscribeJobsModalButton.test.jsx]: yep

File: ui/src/jobs/components/SubscribeJobsModalButton.jsx (hunk 3 above)
Comment (harunurhan) [reply ui/src/jobs/components/SubscribeJobsModalButton.jsx]: 3rd party support. also I imagine we won't have many invalid emails (format wise) for a manual subscription service.

**Pull request merged**